src/SonarQube.Client/Services/SonarQubeClient.cs

```diff
@@ -0,0 +1,239 @@
+                                            ComponentRequest request, CancellationToken token)
+        {
+            const string SearchProjectsAPI = "api/components/search_projects"; // Since 6.2; internal
+
+            var query = AppendQueryString(SearchProjectsAPI, "?organization={0}&p={1}&ps={2}&asc=true",
+                        request.OrganizationKey, request.Page, request.PageSize); // TODO: should handle optional params
+
+            return InvokeSonarQubeApi(connection, query, token,
+                stringResponse => JObject.Parse(stringResponse)["components"].ToObject<ComponentDTO[]>());
+        }
+
+        public Task<Result<OrganizationDTO[]>> GetOrganizationsAsync(ConnectionDTO connection,
+            OrganizationRequest request, CancellationToken token)
+        {
+            const string OrganizationsAPI = "api/organizations/search"; // Since 6.2; internal
+
+            var query = AppendQueryString(OrganizationsAPI, "?p={0}&ps={1}", request.Page, request.PageSize); // TODO: should handle optional params
+
+            return InvokeSonarQubeApi(connection, query, token,
+                stringResponse => JObject.Parse(stringResponse)["organizations"].ToObject<OrganizationDTO[]>());
+        }
+
+        public Task<Result<PluginDTO[]>> GetPluginsAsync(ConnectionDTO connection, CancellationToken token)
+        {
+            const string ServerPluginsInstalledAPI = "api/updatecenter/installed_plugins"; // Since 2.10; internal
+
+            return InvokeSonarQubeApi(connection, ServerPluginsInstalledAPI, token,
+                stringResponse => ProcessJsonResponse<PluginDTO[]>(stringResponse, token));
+        }
+
+        public Task<Result<ProjectDTO[]>> GetProjectsAsync(ConnectionDTO connection, CancellationToken token)
+        {
+            const string ProjectsAPI = "api/projects/index"; // Since 2.10
+
+            return InvokeSonarQubeApi(connection, ProjectsAPI, token,
+                stringResponse => ProcessJsonResponse<ProjectDTO[]>(stringResponse, token));
+        }
+
+        public Task<Result<PropertyDTO[]>> GetPropertiesAsync(ConnectionDTO connection, CancellationToken token)
+        {
+            const string PropertiesAPI = "api/properties/"; // Since 2.6
+
+            return InvokeSonarQubeApi(connection, PropertiesAPI, token,
+                stringResponse => ProcessJsonResponse<PropertyDTO[]>(stringResponse, token));
+        }
+
+        public Task<Result<QualityProfileChangeLogDTO>> GetQualityProfileChangeLogAsync(ConnectionDTO connection,
+            QualityProfileChangeLogRequest request, CancellationToken token)
+        {
+            const string QualityProfileChangeLogAPI = "api/qualityprofiles/changelog"; // Since 5.2
+
+            var query = AppendQueryString(QualityProfileChangeLogAPI, "?profileKey={0}&ps={1}",
+                        request.QualityProfileKey, request.PageSize); // TODO: should handle optional params
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this 'TODO' comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS1135)